### PR TITLE
Docker-based development environment

### PR DESCRIPTION
**Pull request by danguita**
This PR provides Docker based development environment starting point as an alternative to the current one.

Summary of changes:

- Manage containers via Docker Compose.
- Make any configuration environment dependent.
- Add some starting `/script` commands.
- Add MailCatcher container and configuration.
- Add proper docs.

### CODE REVIEW

**Comment by danguita** [comment]
Taking care of conflicts 🙏

**Comment by danguita** [comment]
Green light 🚀

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM ruby:2.3.1
+MAINTAINER Populate <dev@populate.tools>
```
Comment (danguita) [inline Dockerfile]: @ferblape Which would be a good maintainer name and email?

Comment (ferblape) [inline Dockerfile]: I'd say `Populate <lets@populate.tools>`

Comment (danguita) [inline Dockerfile]: 👌 Let me take care.

Comment (danguita) [inline Dockerfile]: These changes are squashed already.

**Comment by ferblape** [comment]
👍

**Pull request merged**